code/jobs/DeployJob.php

```diff
@@ -124,8 +145,14 @@ public function onFailure(Exception $exception) {
 	protected function updateStatus($status) {
 		global $databaseConfig;
 		DB::connect($databaseConfig);
-		$dnDeployment = DNDeployment::get()->byID($this->args['deploymentID']);
-		$dnDeployment->getMachine()->apply($status);
+		$deployment = DNDeployment::get()->byID($this->args['deploymentID']);
+		$deployment->getMachine()->apply($status);
+
+		$backup = $deployment->BackupDataTransfer();
+		if($backup && $backup->exists()) {
+			$backup->Status = $status;
```

Comment:
Watch out, `Status` is different to `State`. This function gets passed the `State`, but is named to fit with the old API. The "$status" should probably be renamed to "$state". You might need to map from `DNDeployment::TR_COMPLETE` to Backup::Status